packages/worker/src/worker-options.ts

```diff
@@ -411,13 +416,100 @@ export function addDefaultWorkerOptions(options: WorkerOptions): WorkerOptionsWi
     // 4294967295ms is the maximum allowed time
     isolateExecutionTimeout: debugMode ? '4294967295ms' : '5s',
     workflowThreadPoolSize: 8,
-    maxCachedWorkflows: maxCachedWorkflows ?? Math.max(os.totalmem() / GiB - 1, 1) * 200,
+    maxCachedWorkflows:
+      maxCachedWorkflows ??
+      Math.floor(Math.max((systemResources.memory + systemResources.swap - systemResources.heap) / GiB - 1, 1) * 250),
     enableSDKTracing: false,
     debugMode: debugMode ?? false,
     interceptors: appendDefaultInterceptors({}),
     sinks: defaultSinks(),
     ...rest,
   };
+
+  reportWorkerOptions(optionsWithDefaults);
```

Comment:
I'd report during worker construction and use the worker's logger (even though it's the same as the runtime logger), when you call `Runtime.instance()` you're effectively instantiating a runtime which would prevent the user from installing a runtime with their own settings.